src/gtk/window.cpp

```diff
@@ -3709,6 +3710,44 @@ touch_callback(GtkWidget* widget, GdkEventTouch* gdk_event, wxWindow* win)
     if ( !data )
         return;
 
+    if ( data->m_rawTouchEvents)
+    {
+        wxEventType type;
+
+        switch(gdk_event->type)
+        {
+        case GDK_TOUCH_BEGIN:
+            type = wxEVT_TOUCH_BEGIN;
+            break;
+
+        case GDK_TOUCH_UPDATE:
+            type = wxEVT_TOUCH_MOVE;
+            break;
+
+        case GDK_TOUCH_END:
+            type = wxEVT_TOUCH_END;
+            break;
+
+        case GDK_TOUCH_CANCEL:
+            type = wxEVT_TOUCH_CANCEL;
+            break;
+
+        default:
+            type = wxEVT_NULL;
+        }
+        if (type != wxEVT_NULL)
+        {
+            wxMultiTouchEvent event(win->GetId(), type);
+
+            event.SetEventObject(win);
+            event.SetPosition(wxPoint(gdk_event->x, gdk_event->y));
+            event.SetSequenceId(wxTouchSequenceId(gdk_event->sequence));
+            event.SetPrimary(gdk_event->emulating_pointer);
+
+            win->GTKProcessEvent(event);
```

Comment:
Is it correct to ignore its return value here?

Reply:
Should we still call `wxEmulateLeftDownEvent`/`wxEmulateMotionEvent`/`wxEmulateLeftUpEvent`, even if the touch event was handled?

Note that MSW sends synthesized mouse events regardless.